app/src/Talk/TalkController.php

```diff
@@ -475,15 +480,21 @@ public function unlinkSpeaker($eventSlug, $talkSlug, $username)
 
         $unlinkSpeakerUri = $talkUri . "/speakers/" . $userId;
 
-        try {
-            $talkApi->unlinkVerifiedSpeakerFromTalk($unlinkSpeakerUri);
-        } catch (Exception $e) {
-            $this->application->flash('error', $e->getMessage());
+        $isAdmin = $event->getCanEdit();
+        if (!$isAdmin && $talk->getStartDateTime()->format('U') > time()) {
+            try {
+                $talkApi->unlinkVerifiedSpeakerFromTalk($unlinkSpeakerUri);
+            } catch (Exception $e) {
+                $this->application->flash('error', $e->getMessage());
+                $this->application->redirect($url);
+            }
+
+            $this->application->flash('message', 'Speaker has been removed from this talk.');
+            $this->application->redirect($url);
```

Comment:
The redirect throws a `Stop` Exception so may not be needed, though returning the value makes it more clear